src/components/AddPlayerForm.js

```diff
@@ -1,45 +1,100 @@
 import { React } from 'react';
 // directly access to reference cue DOM elements
-import { useRef } from 'react';
 import { Form, Button, Spinner } from 'react-bootstrap';
 
+import useInput from '../hooks/use-input'
+
 function AddPlayerForm(props) {
-    const nameInputRef = useRef();
-    const emailInputRef = useRef();
+    // use hook useInput
+    const {
+        value: enteredName,
+        isValid: enteredNameIsValid,
+        hasError: nameInputHasError,
+        valueChangeHandler: nameChangeHandler,
+        inputBlurHanlder: nameBlurHanlder,
+        reset: resetNameInput
+    } = useInput(value => value.trim() !== '' && value.length >= 2);
 
-    // disable button in case there's an http request running
-    // use the useEffect hook to set a state defalt: false
-    
+    const {
+        value: enteredEmail,
+        isValid: enteredEmailIsValid,
+        hasError: EmailInputHasError,
+        valueChangeHandler: emailChangeHandler,
+        inputBlurHanlder: emailBlurHanlder,
+        reset: resetEmailInput
+    } = useInput(value => value.includes('@'));
+
+    // to manage overall form validity
+    let formIsValid = false;
+
+    if (enteredNameIsValid && enteredEmailIsValid) {
+        formIsValid = true;
+    }
 
     const submitHandler = (event) => {
-        // prevent the browser default to allow to fully handle the submission
+        // prevent the browser default of senfing a request,  to allow to fully handle the submission
         event.preventDefault();
-        
-        const enteredName = nameInputRef.current.value;
-        const enteredEmail = emailInputRef.current.value;
-     
+
+        if (!formIsValid) {
+            return;
+        }
+
+        // if enteredNameIsValid is valid, just return and stop here
+        // if (props.showNameErrorMessage) {
+        //     console.log('showNameErrorMessage true', props.showNameErrorMessage)
+        //     // return;
+        // } else {
+        //     console.log('showNameErrorMessage false', props.showNameErrorMessage)
+        // }
+             
         // Rest API to send server & DB
         const playerData = {
             name: enteredName,
             email: enteredEmail
         };
-   
-        props.onAddPlayer(playerData) 
-    }
+
+        props.onAddPlayer(playerData);
+
+        // reset states
+        resetNameInput();
```

Comment:
I find this behaviour not very kind. When the user makes mistake, they have to type from the beginning. We might lose one player if they give up without the second try.. How about keeping the input so they can fix it since the error message is served nicely.